### ISSUE TITLE
Use .github?

### ISSUE THREAD

**Issue by cobraz** [pulumiverse/community#15]
👋  

Do we wanna use [the special `.github` repository](https://stackoverflow.com/a/61301254)?

In that case, do we still wanna use this repository to maintain the files? E.g. `CONTRIBUTING.md` could be stored in this repository, and pull requests opened here, but we could have a Github Workflow that updates the files (less overhead).

Thoughts?

**Comment by ringods** [comment]
@cobraz is there any official Github documentation on `.github` as a repository? I was unaware of this feature and would like to read more about it.

**Comment by simenandre** [comment]
I didn't find any. What do you need? 🙂

**Comment by ringods** [comment]
@cobraz curiosity. If I could read in official Github documentation what all we could do with that special repo, we can generate ideas for our org. 🤷🏼‍♂️

**Comment by ringods** [comment]
@cobraz I think I found the official docs around this:

https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file

Especially the following paragraph is of interest:

> For example, anyone who creates an issue or pull request in a repository that does not have its own CONTRIBUTING file will see a link to the default CONTRIBUTING file. If a repository has any files in its own `.github/ISSUE_TEMPLATE` folder, including issue templates or a *config.yml* file, none of the contents of the default `.github/ISSUE_TEMPLATE` folder will be used.

This allows to not having to duplicate the community files in every repository. So, I'm all in to migrate the contents of this repo to the `.github` repository.

**Comment by simenandre** [comment]
What do you think about where the "source of truth" should be?

**Comment by ringods** [comment]
`.github`!

Ok if I migrate everything over to that repo and remove this one?

**Comment by simenandre** [comment]
I'm not sure we should. There are files that aren't necessarily intuitive to find in `.github`. Like the governance files/policies etc.

Maybe it makes more sense to have multiple repositories; e.g. one for policies? 🤷‍♂️

Let's dig a bit more on this?

@ringods why do you want to use `.github`?

**Comment by ringods** [comment]
@cobraz I'm puzzled. You suggested to start using `.github` by opening this ticket. When I agree with you, you ask back why I want to use it.

Let me rephrase why:

> This allows to not having to duplicate the community files in every repository. So, I'm all in to migrate the contents of this repo to the .github repository.

**Comment by simenandre** [comment]
> @cobraz I'm puzzled. You suggested to start using `.github` by opening this ticket. When I agree with you, you ask back why I want to use it.

To clarify, I think we should use the repository, but I'm unsure if `.github` is where we should store the source of the documents, accept and review pull requests etc. This repository (`community`) is more intuitive in its naming, developers are less likely to misinterpret the contents it. But, having `.github` be source of truth means less overhead.

Again, if we decide to keep both and have `community` have all documents, we can have a simple sync between the two (exclusively for GitHub-related documents, such as CODE_OF_CONDUCT, CONTRIBUTING, SECURITY). Meaning, if we update specific files in `community` (like CONTRIBUTING) it automatically is committed and pushed to `.github`.

**Comment by ringods** [comment]
@cobraz I took the liberty to make some changes as to generate the screenshot below. Having the file `profile/README.md` front and center makes for a nice welcome, which we can adapt to our likings. Any relative link in that document (e.g. issues) points to the `.github` repository.

Second, I pinned the `.github` repository and adapted the "About" text for that repository. With a clear "About" text, it should be obvious what the repo is for.

Third, why bother setting up a sync scheme just because the name of the repository isn't intuitive. With what I did in previous steps, would that solve your concerns?

If you would agree to using `.github`, I will take it upon me to transfer all issues to that repo to not loose any context.

<img width="1035" alt="Screenshot 2021-11-17 at 21 03 07" src="https://user-images.githubusercontent.com/77923/142274193-f873412d-4470-408a-bd39-4d4116d472d3.png">

**Comment by simenandre** [comment]
> @cobraz I took the liberty to make some changes as to generate the screenshot below. Having the file `profile/README.md` front and center makes for a nice welcome, which we can adapt to our likings. Any relative link in that document (e.g. issues) points to the `.github` repository.

Yes, that looks good! 👍

> Second, I pinned the `.github` repository and adapted the "About" text for that repository. With a clear "About" text, it should be obvious what the repo is for.
> ...
> If you would agree to using `.github`, I will take it upon me to transfer all issues to that repo to not loose any context.

It's probably fine for now. We're not making permanent decisions here. But, IMHO, this feels/looks _hacky_.

Having thought about it for a little while, I'm leaning more towards having a second repository for policies (e.g. `pulumiverse/policies`) and point people around with our README (located in `.github`). Probably with only Github-related documents in `.github`. That said, you can migrate to `.github` for now if you'd like. 👍

**Comment by ringods** [comment]
Please no. Then we loose all the power of centrally managing these files and have the links immediately in all repositories. Here is an example:

<img width="483" alt="Screenshot 2021-11-17 at 21 33 57" src="https://user-images.githubusercontent.com/77923/142278337-1a2a774d-9f60-4d56-9e72-d121fd06ec72.png">

This is a part of the webpage of [awesome-pulumi](https://github.com/pulumiverse/awesome-pulumi). The link to the `Code of Conduct` goes straight to [`.github/CODE_OF_CONDUCT.md`](https://github.com/pulumiverse/.github/blob/main/CODE_OF_CONDUCT.md). You won't have that if you manage these files in another repository.

Looking forward, I only expect Github to offer more power to come of this setup.

**Comment by simenandre** [comment]
I'm not talking about `.github/CODE_OF_CONDUCT.md`, as that is a Github-related file (since it has the functionality you're referring to above).

We are going to need more documentation than what is related to the Github organization.

That said, let's move forward. I'm not saying I'm putting a vote forward for creating a policies repository right now, I think we should move forward – and talk more about this down the line.

**Comment by ringods** [comment]
OK, I'll unify to `.github` for now.
